components/core/src/output.rs

```diff
@@ -154,59 +270,13 @@ impl<'a> Serialize for StructuredOutput<'a> {
     }
 }
 
-// If we ever want to create multiple output formats in the future, we would do it here -
-// essentially create a flag we check to see what output you want, then call a different formatting
-// function. Viola!
 impl<'a> fmt::Display for StructuredOutput<'a> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        if self.json.unwrap_or_else(is_json) {
-            // Our JSON serialization handles verbosity itself, and
-            // color is ignored anyway, so there's no reason to check
-            // those settings here.
-
-            // unwrap is safe, as we control the inputs
-            let as_json = serde_json::to_string(&self).unwrap();
-            write!(f, "{}", as_json)
-        } else {
-            let verbose = self.verbose.unwrap_or_else(is_verbose);
-            let color = self.color.unwrap_or_else(is_color);
-
-            let preamble_color = if self.preamble == PROGRAM_NAME.as_str() {
-                Cyan
-            } else {
-                Green
-            };
-
-            if verbose {
-                if color {
-                    write!(
-                        f,
-                        "{}({})[{}]: {}",
-                        preamble_color.paint(self.preamble),
-                        White.bold().paint(self.logkey),
-                        White
-                            .underline()
-                            .paint(format!("{}:{}:{}", self.file, self.line, self.column)),
-                        self.content
-                    )
-                } else {
-                    write!(
-                        f,
-                        "{}({})[{}:{}:{}]: {}",
-                        self.preamble, self.logkey, self.file, self.line, self.column, self.content
-                    )
-                }
-            } else if color {
-                write!(
-                    f,
-                    "{}({}): {}",
-                    preamble_color.paint(self.preamble),
-                    White.bold().paint(self.logkey),
-                    self.content
-                )
-            } else {
-                write!(f, "{}({}): {}", self.preamble, self.logkey, self.content)
-            }
+        let bufwtr = BufferWriter::stdout(self.color_choice());
+        let mut buffer = bufwtr.buffer();
+        match self.format(&mut buffer) {
+            Ok(_) => f.write_str(&String::from_utf8_lossy(buffer.as_slice())),
```

Comment:
Are we certain that `buffer` will contain valid utf8? If so, why not
```suggestion
            Ok(_) => f.write_str(std::str::from_utf8(buffer.as_slice()).expect("termcolor buffer valid utf8")),
```
And if we're not, how might we be getting invalid utf8 in there?